bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/internal/SecureHttpClientFactory.java

```diff
@@ -0,0 +1,278 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.net.http.internal;
+
+import java.security.AccessController;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+import java.util.Map;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.eclipse.jetty.util.thread.QueuedThreadPool;
+import org.eclipse.smarthome.io.net.http.CommonHttpClient;
+import org.eclipse.smarthome.io.net.http.HttpClientFactory;
+import org.eclipse.smarthome.io.net.http.TrustManagerProvider;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Factory class to create jetty http clients
+ *
+ * @author Michael Bock - initial API
+ */
+@Component(service = HttpClientFactory.class, immediate = true, configurationPid = "org.eclipse.smarthome.HttpClientFactory")
+public class SecureHttpClientFactory implements HttpClientFactory {
+
+    private final Logger logger = LoggerFactory.getLogger(SecureHttpClientFactory.class);
+
+    private static final String CONFIG_MIN_THREADS_SHARED = "minThreadsShared";
+    private static final String CONFIG_MAX_THREADS_SHARED = "maxThreadsShared";
+    private static final String CONFIG_KEEP_ALIVE_SHARED = "keepAliveTimeoutShared";
+    private static final String CONFIG_MIN_THREADS_CUSTOM = "minThreadsCustom";
+    private static final String CONFIG_MAX_THREADS_CUSTOM = "maxThreadsCustom";
+    private static final String CONFIG_KEEP_ALIVE_CUSTOM = "keepAliveTimeoutCustom";
+
+    private static final int MIN_CONSUMER_NAME_LENGTH = 4;
+    private static final int MAX_CONSUMER_NAME_LENGTH = 20;
+    private static final Pattern CONSUMER_NAME_PATTERN = Pattern.compile("[a-zA-Z0-9_\\-]*");
+
+    private volatile TrustManagerProvider trustmanagerProvider;
+
+    private QueuedThreadPool threadPool = null;
```

Comment:
`null` is the default value imho it does not make much sense to assign it explicitly here.

Reply:
Right